### PR TITLE
add maven 3.6.3

### PR DESCRIPTION
**Pull request by Pro**
Specify library name and version:  **apache-maven/3.6.3**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/apache-maven/3.6.3/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import os
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+
+
+class PackageApacheMaven(ConanFile):
+    name = "apache-maven"
```
Comment (Croydon) [inline recipes/apache-maven/3.6.3/conanfile.py:7]: I think that we might want to call this just maven. At least I don't know of any other popular "maven"

```suggestion
    name = "maven"
```

File: recipes/apache-maven/3.6.3/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    version = "3.6.3"
```
Comment (Croydon) [inline recipes/apache-maven/3.6.3/conanfile.py:8]: Don't hardcode the version here

```suggestion
```

As a consequence also rename the folder name from `3.6.3` to `all`

Making individual folders/recipes is only needed when the recipe differs strongly from version to version

File: recipes/apache-maven/3.6.3/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Apache Maven is a software project management and comprehension tool"
+    license = ("Apache-2.0")
+    homepage = "https://maven.apache.org/"
+    topics = ("conan", "maven", "java")
+    settings = "os", "arch", "compiler"
+
+    requires = ("zulu-openjdk/[>8.0]")
```
Comment (Croydon) [inline recipes/apache-maven/3.6.3/conanfile.py:16]: Version ranges are not allowed

File: recipes/maven/all/conandata.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "3.6.3":
+    url: https://downloads.apache.org/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.tar.gz
```
Comment (prince-chrismc) [inline recipes/maven/all/conandata.yml:3]: I took a peak the source code is available...

https://downloads.apache.org/maven/maven-3/3.6.3/source/apache-maven-3.6.3-src.tar.gz

Im curious what would be the requirements for building this

**Comment by prince-chrismc** [comment]
Why do you need this available through conan? it seems like a very easy project to get from system package managers?
(curiosity, what other use cases exist?)

File: recipes/maven/all/conandata.yml (hunk 4 above)
Comment (Pro) [reply recipes/maven/all/conandata.yml:3]: I also did not try to build from source, but I guess that it is quite large, e.g. the following site warns that the full source build builds 400 modules... And then the question is, which modules should we build here?

https://maven.apache.org/scm.html

**Comment by Pro** [comment]
> Why do you need this available through conan? it seems like a very easy project to get from system package managers?
> (curiosity, what other use cases exist?)

Good question :)
I guess many projects in the conan center could be installed through system package managers such as apt or pip.

Please note that I'm not (yet) a conan expert, when reading the following lines:

As far as I understood, conan should be used for dependency management of various application components.
At our company we have the goal/dream to manage all packages via conan, i.e., just install a plain and clean ubuntu with some necessary applications, and provide all the (command line) tooling and development libraries through conan.
Also, our own developed components are managed via conan. With this we aim to avoid sayings of our developers like "Hmm, but it works on my PC, why isn't it working on yours".

This has the big advantage that every developer has the same requirements installed with exactly the same version.

For the specific case of maven, we are currently testing Cyclone DDS. Since it should be used in our C++ applications as dependency, it should come through Conan. See https://github.com/eclipse-cyclonedds/cyclonedds/pull/686

To build Cyclone DDS, it requires maven for some source code generation (more specifically, to generate C++ code out of IDL files). Also for later usage in your CMake pipeline, you will need at least Java, probably also Maven to go from IDL to C++.

Yes, you could install maven with apt, and then build the conan package for Cyclone DDS, but then you would derivate from the clean base system, where all build dependencies are managed through conan.

**Comment by prince-chrismc** [comment]
My team moved to Conan to be able to leverage OSS and I can relate to the dream 🤗  

> To build Cyclone DDS, it requires maven for some source code generation (more specifically, to generate C++ code out of IDL files). 

That's a very strong motivation, but I am wondering if the same effect could be obtained with a system package. (There's a few examples in CCI).

The binaries from the distribution are not portable so you might just end up with a tool that does not work because of `glibc` issues.

File: recipes/maven/all/conandata.yml (hunk 4 above)
Comment (Croydon) [reply recipes/maven/all/conandata.yml:3]: Are all of those modules in this binary download?

**Comment by Croydon** [comment]
System packages are a nuclear option for the worst cases, please don't advertise them 😄

File: recipes/maven/all/conandata.yml (hunk 4 above)
Comment (Pro) [reply recipes/maven/all/conandata.yml:3]: That's a good question which I cannot answer for sure... I couldn't really find out which modules are included in the bin download.
My guess is that the bin download only contains the necessary basis.

**Comment by prince-chrismc** [comment]
What is in the boot and conf folders?

**Pull request closed without merging**